lib/ProductOpener/Ingredients.pm

```diff
@@ -4298,6 +4396,20 @@ sub preparse_ingredients_text($$) {
 	# turn & to and
 	$text =~ s/ \& /$and/g;
 
+	# number + gr / grams -> g 
+	$text =~ s/(\d\s*)(gr|gram|grams)\b/$1g/ig;
+	if ($product_lc eq 'fr') {
+		$text =~ s/(\d\s*)(gramme|grammes)\b/$1g/ig;
+	}
```

Comment:
would'nt it be safer to do this the other way around ?

I mean replacing `g` / `gr` with `grammes`

Reply:
If we replace g by grammes, then we have to make a lot of code that analyze the resulting text language specific, which increases the complexity of the parsing (which is already quite complex). That's why I think it's best to normalize to "g".